? msw_bitmap2image_transparency.patch
? gtk_bitmap2image_transparency.patch
Index: src/msw/bitmap.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/bitmap.cpp,v
retrieving revision 1.100
diff -r1.100 bitmap.cpp
829a830,836
>     // the colour used as transparent one in wxImage and the one it is replaced
>     // with when it really occurs in the bitmap
>     static const int MASK_RED = 1;
>     static const int MASK_GREEN = 2;
>     static const int MASK_BLUE = 3;
>     static const int MASK_BLUE_REPLACEMENT = 2;
> 
921,922c928,929
<         // background color set to RGB(16,16,16) in consistent with wxGTK
<         unsigned char r=16, g=16, b=16;
---
>         // background color set to mask colour
>         unsigned char r=MASK_RED, g=MASK_GREEN, b=MASK_BLUE;
929a937,939
>                 {
>                     if( (ptdata[0] == r) && (ptdata[1] == g) && (ptdata[2] == b) )
>                         ptdata[2] = MASK_BLUE_REPLACEMENT;
930a941
>                 }

 	  	 
